### PR TITLE
Improve mesh refinement documentation

### PR DESCRIPTION
**Pull request by jorgensd**

### CODE REVIEW

File: cpp/dolfinx/refinement/plaza.cpp [hunk 1]
```diff
@@ -24,13 +24,22 @@ using namespace dolfinx::refinement;
 namespace
 {
 //-----------------------------------------------------------------------------
-// 2D version of subdivision allowing for uniform subdivision (flag)
+/// 2D version of subdivision allowing for uniform subdivision (flag)
+/// @param[in] indices Vector of size (num_vertices + num_edges) containing the
```
Comment (garth-wells) [inline cpp/dolfinx/refinement/plaza.cpp:28]: Open with what at is, then give size.

File: cpp/dolfinx/refinement/plaza.cpp [hunk 2]
```diff
@@ -24,13 +24,22 @@ using namespace dolfinx::refinement;
...
+/// global indices for the original vertices and potential new vertices at each
+/// edge. If an edge is not refined its corresponding entry is -1.
+/// @param[in] longest_edge Vector indicating the longest edge for each
```
Comment (garth-wells) [inline cpp/dolfinx/refinement/plaza.cpp:31]: It's not a vector, it's an integer.

File: cpp/dolfinx/refinement/plaza.cpp [hunk 3]
```diff
@@ -24,13 +24,22 @@ using namespace dolfinx::refinement;
...
+///   triangle. For tdim=2, one entry, for tdim=3, four entries. The entry
+///   relates to the local index of the last num_edges entries of indices.
+/// @returns Local indices for each sub-divived triangle
```
Comment (garth-wells) [inline cpp/dolfinx/refinement/plaza.cpp:34]: The boolean arg is mising.

File: cpp/dolfinx/refinement/plaza.h [hunk 4]
```diff
@@ -396,7 +430,7 @@ compute_refinement(MPI_Comm neighbor_comm,
 
       // Convert from cell local index to mesh index and add to cells
       for (std::int32_t v : simplex_set)
-        cell_topology.push_back(indices[v]);
+        cell_topology.push_back([v]);
```
Comment (jorgensd) [inline cpp/dolfinx/refinement/plaza.h:433]:
```suggestion
        cell_topology.push_back(indices[v]);
```

File: cpp/dolfinx/refinement/plaza.cpp [hunk 5]
```diff
@@ -24,13 +24,23 @@ using namespace dolfinx::refinement;
...
 //-----------------------------------------------------------------------------
-// 2D version of subdivision allowing for uniform subdivision (flag)
+/// 2D version of subdivision allowing for uniform subdivision (flag)
+/// @param[in] indices Vector containing the
+/// global indices for the original vertices and potential new vertices at each
+/// edge. If an edge is not refined its corresponding entry is -1. Size
+/// `num_vertices + num_edges`
+/// @param[in] longest_edge Local index of the longest edge in the triangle.
+/// @param[in] uniform If true, the triangle is subdivided into four
+/// sub-triangles.
```
Comment (chrisrichardson) [inline cpp/dolfinx/refinement/plaza.cpp:34]:
```suggestion
/// @param[in] uniform If true, the triangle is subdivided into four similar
/// sub-triangles.
```

File: cpp/dolfinx/refinement/plaza.cpp [hunk 6]
```diff
@@ -24,13 +24,23 @@ using namespace dolfinx::refinement;
 namespace
 {
 //-----------------------------------------------------------------------------
-// 2D version of subdivision allowing for uniform subdivision (flag)
+/// 2D version of subdivision allowing for uniform subdivision (flag)
+/// @param[in] indices Vector containing the
+/// global indices for the original vertices and potential new vertices at each
+/// edge. If an edge is not refined its corresponding entry is -1. Size
+/// `num_vertices + num_edges`
```
Comment (chrisrichardson) [inline cpp/dolfinx/refinement/plaza.cpp:31]:
```suggestion
/// @param[in] indices Vector of size `num_vertices + num_edges` containing new
/// global vertex indices for each point. If an edge is not refined its corresponding entry is -1.
/// 
```

File: cpp/dolfinx/refinement/plaza.h [hunk 7]
```diff
@@ -116,10 +116,13 @@ compute_parent_facets(std::span<const std::int32_t> simplex_set)
...
-/// @param[in] indices Vector indicating which edges are to be
-///   split (value >=0)
+/// @param[in] indices Vector containing the global indices for the original
+/// vertices and potential new vertices at each edge. Size (num_vertices +
+/// num_edges).
+//// If an edge is not refined its corresponding entry is -1
 /// @param[in] longest_edge Vector indicating the longest edge for each
-///   triangle. For tdim=2, one entry, for tdim=3, four entries.
+///   triangle. For tdim=2, one entry, for tdim=3, four entries. The entry
+///   relates to the local index of the last num_edges entries of indices.
```
Comment (chrisrichardson) [inline cpp/dolfinx/refinement/plaza.h:125]:
```suggestion
///   triangle in the cell. For triangular cells (2D) there is only one value, and for tetrahedra (3D) there are four values, one for each facet. The values give the local edge indices of the cell.
```

File: cpp/dolfinx/refinement/plaza.h [hunk 8]
```diff
@@ -255,7 +266,30 @@ face_long_edge(const mesh::Mesh<T>& mesh)
...
+/// @param[in] long_edge A map from each face to its longest edge. Index is
+/// local to the process.
+/// @param[in] edge_ratio_ok For each face in a 2D mesh this error contains a
+/// marker indicating if the ratio between smallest and largest edge is bigger
+/// than sqrt(2)/2
+/// @param[in] option Option to compute additional information relating refined
+/// and original mesh entities
+/// @returns (0) The new mesh topology, (1) the new flattened mesh geometry, (3)
+/// Shape of the new geometry_shape, (4) Map from new cells to parent cells
+/// and (5) map from refined facets to parent facets.
```
Comment (chrisrichardson) [inline cpp/dolfinx/refinement/plaza.h:292]: Should say which of these are optional outputs

File: cpp/dolfinx/refinement/plaza.cpp (hunk 6 above)
Comment (jorgensd) [reply cpp/dolfinx/refinement/plaza.cpp:31]: I changed this after feedback from Garth (https://github.com/FEniCS/dolfinx/pull/3010#discussion_r1462384505)

**Review by chrisrichardson — APPROVED** [review]

**Comment by jhale** [comment]
Please merge this when you are happy with it.

**Pull request merged**